alica_tests/src/test/test_alica_simple_plan_tree_serialization.cpp

```diff
@@ -0,0 +1,71 @@
+#include "test_alica.h"
+
+#include "engine/SimplePlanTree.h"
+#include "engine/containers/PlanTreeInfo.h"
+#include <alica/test/Util.h>
+#include <engine/RunningPlan.h>
+
+#include <gtest/gtest.h>
+
+namespace alica
+{
+
+class AlicaSerializationTest : public AlicaTestFixture
+{
+protected:
+    const char* getRoleSetName() const override { return "Roleset"; }
+    const char* getMasterPlanName() const override { return "SimpleTestPlan"; }
+    bool stepEngine() const override { return false; }
+
+    static constexpr long SimpleTestPlanId = 1412252439925;
+    static constexpr long TestState2Id = 1412761855746;
+
+    void waitUntilPlan(long planId)
+    {
+        alica::AlicaTime sleepTime = alica::AlicaTime::seconds(1);
+        do {
+            ae->getAlicaClock().sleep(sleepTime);
+        } while (!alica::test::Util::isPlanActive(ae, SimpleTestPlanId));
+    }
+
+    // Calls private methods inside TeamObserver
+    std::unique_ptr<SimplePlanTree> serializeAndDeserialize(IdGrp msg_to_send)
+    {
+        // Serialize (see TeamObserver::doBroadCast)
+        PlanTreeInfo pti = ae->getTeamObserver().sptToMessage(msg_to_send);
+
+        // De-serialize (see TeamObserver::updateTeamPlanTrees and TeamObserver::handlePlanTreeInfo)
+        std::unique_ptr<SimplePlanTree> spi = ae->getTeamObserver().sptFromMessage(pti.senderID, pti.dynamicStateIDPairs, ae->getAlicaClock().now());
+        return spi;
+    }
+};
+
+/**
+ * Tests whether plan serialization works fine
+ */
+TEST_F(AlicaSerializationTest, serializeDeserialize)
```

Comment:
need to add more comprehensive test cases. The tree should have atleast 2-3 levels with multiple children per node & try a few different tree structures

Reply:
Test isn't passing still, so I left it commented until feature is implemented

Reply:
yeah, I don't think the deserialization will work as it is right now